func/internal/podevaluator.go

```diff
@@ -625,15 +647,74 @@ func (pm *podManager) getImageMetadata(ctx context.Context, ref name.Reference,
 	return de, nil
 }
 
+func getImage(ctx context.Context, ref name.Reference, auth authn.Authenticator, image string, enablePrivateRegistries bool, enablePrivateRegistriesTls bool, tlsSecretPath string) (containerregistry.Image, error) {
+	if enablePrivateRegistries && !strings.HasPrefix(image, defaultRegistry) && enablePrivateRegistriesTls {
```

Comment:
Should the flow of control really fall into the `else` if any of these conditions are false?

Reply:
i believe so. the only difference in the if vs the else is that it attempts to use the given tls cert for connecting to the registry to pull the image instead of the "well-known" websites certs present on the cluster. in the case where private registries are not enabled or the registry name given is the default reg (gcr/kpt-fn/...) or if the custom tls is not enabled for these registries we should not be using a custom tls cert for pulling an image.

Reply:
Thanks @Catalin-Stratulat-Ericsson

Reply:
changed priority in the if statement as requested.